### DISCUSSION TITLE
Reducing the CI burden of rust-mozjs and mozjs

### DISCUSSION THREAD

**Discussion by jdm** [servo/servo#27455] [General]
Every time we make a pull request to servo/rust-mozjs or servo/mozjs (or push a new revision, or approve a PR, etc.) we throw a wrench into our CI queue. Both of these repositories test on nightly, beta, and stable, and they test both the debugmozjs configuration and the default configuration, and they also test on both linux and macOS. This means that there are 12 different CI tasks queued; each job takes between 30 and 60 minutes, and our travisCI tier only lets us run 5 jobs in parallel across all our repositories, so this means any other repositories trying to merge are stuck in the CI queue for several hours before we get any results from them.

I'd like to reduce the burden here by removing the beta configuration entirely (so only testing on nightly and stable), and only testing debug against stable. This would reduce us from 12 tasks to 6 and still give us coverage over the most important facets of the work in my opinion.

Do you see any alternatives that would provide better coverage? Am I missing anything?

**Comment by SimonSapin** [comment]
> removing the beta configuration entirely (so only testing on nightly and stable), and only testing debug against stable

Sounds good.

Merging the repositories would also allow them to share CI, reducing the total number of jobs needed to make changes in `servo/mozjs` and use them in `servo/servo`.

**Reply by jdm** [reply]
Oh yeah, that would indeed be a big improvement.
